packages/suite/src/views/suite/firmware/index.tsx

```diff
@@ -0,0 +1,158 @@
+import React from 'react';
+import { connect } from 'react-redux';
+import { bindActionCreators } from 'redux';
+import styled from 'styled-components';
+// import { FormattedMessage } from 'react-intl';
+
+import { Button, P, H1, H4, H5, variables } from '@trezor/components';
+import { goto } from '@suite-actions/routerActions';
+import { getRoute } from '@suite/utils/suite/router';
+import { firmwareUpdate } from '@suite-actions/firmwareActions';
+import { Dots } from '@suite/components/onboarding/Loaders';
+
+// todo: rework to common notifications
+import WalletNotifications from '@wallet-components/Notifications';
+
+import { AppState } from '@suite-types';
+// import l10nMessages from './index.messages';
+
+const Wrapper = styled.div`
+    display: flex;
+    flex-direction: column;
+    align-items: center;
+    justify-content: center;
+    padding: 60px 24px 30px 24px;
+    flex: 1;
+`;
+
+const Top = styled.div`
+    display: flex;
+    flex-direction: column;
+    max-width: 500px;
+    text-align: center;
+    flex: 1;
+    padding-top: 30px;
+`;
+
+const Middle = styled.div`
+    display: flex;
+    flex: 1;
+`;
+
+const Bottom = styled.div`
+    display: flex;
+    flex-direction: column;
+    flex: 1;
+`;
+
+// todo warnings checkboxes
+// const Checkboxes = styled.div`
+//     flex: 1;
+// `;
+
+const ChangeLog = styled.div`
+    flex: 1;
+`;
+
+const TitleHeader = styled(H1)`
+    display: flex;
+    font-size: ${variables.FONT_SIZE.HUGE};
+    justify-content: center;
+    align-items: center;
+    flex-wrap: wrap;
+`;
+
+interface Props {
+    firmware: AppState['firmware'];
+    device: AppState['suite']['device'];
+    firmwareUpdate: typeof firmwareUpdate;
+    goto: typeof goto;
+}
+
+const FirmwareUpdate = (props: Props) => {
+    const { device, firmware } = props;
+
+    if (!device || device.type !== 'acquired') return null;
+
+    const isInProgress = () => {
+        if (!firmware || !firmware.status) return null;
+        return ['started', 'downloading', 'installing', 'restarting'].includes(firmware.status);
+    };
+
+    const getStatus = () => {
+        if (!device || !device.connected) {
+            return 'no-device';
+        }
+        if (isInProgress()) {
+            return 'in-progress';
+        }
+        if (device.mode === 'bootloader') {
+            return 'in-bl';
+        }
+        if (device.firmware === 'valid') {
+            return 'up-to-date';
+        }
+        if (['outdated', 'required'].includes(device.firmware)) {
+            return 'update-available';
+        }
+    };
+
+    return (
+        <Wrapper>
+            {/* todo: use proper notifications, leaving it here as a starting point for next iteration */}
+            <WalletNotifications />
+            <Top>
+                <TitleHeader>
+                    {/* todo: this does not appear right now, it displays ConnectDevice page. */}
+                    {getStatus() === 'no-device'
```

Comment:
![Screenshot 2019-08-27 at 14 06 43](https://user-images.githubusercontent.com/6961901/63769826-1310c980-c8d4-11e9-87ce-9a54bb215a27.png)

They are not. I am calling for reenabling this rule and banning nested ternary 🚨

Reply:
HAHA :D Well. For example in this case, I can imagine a function, something like `getHeadingTextForStatus(status)`. But what if I have quite complex tree to render based on many conditions. Of course the tree might be binary, but nesting will become quite deep then. Of course, I could create a fucntion `getComponentForStatus(status)` and return entire component based on given conditions. If this is what you like more, why not.

Reply:
Well, I don't fully understand your usecase, but there are whole languages without a ternary operator. I am sure you can find a way to write your code without NESTED ternary op 😀 They are also disabled in eslint by default, for a good reason imo. 
http://golang.org/doc/faq#Does_Go_have_a_ternary_form